hbase-client/src/main/java/org/apache/hadoop/hbase/filter/FuzzyRowFilter.java

```diff
@@ -161,6 +161,31 @@ private boolean isPreprocessedMask(byte[] mask) {
     return true;
   }
 
+  /**
+   * Returns the Fuzzy keys in the format expected by the constructor.
+   * @return the Fuzzy keys in the format expected by the constructor
+   */
+  public List<Pair<byte[], byte[]>> getFuzzyKeys() {
+    List<Pair<byte[], byte[]>> returnList = new ArrayList<>(fuzzyKeysData.size());
+    for (Pair<byte[], byte[]> fuzzyKey : fuzzyKeysData) {
```

Comment:
Should we have a boolean flagging when masks have been preprocessed, so that we could avoid these loops when it hasn't?

Reply:
It doesn't really matter, becase this method is called only once when we serialize the filter on the client side for a REST call, so it is not in the hot path.

I can add the same checks that the preprocessMask() has for symmetry.

Reply:
I have added checks around the revert code.